_commands/toys/cowsay.md

```diff
@@ -2,23 +2,111 @@
 ---
 
 cowsay
--------
+--
+`cowsay` generates an ASCII picture of a cow saying something provided by  the  user.
 
-ToDo: Needs Completion
 
 ~~~ bash
-*short* example of command usage and output
+$ cowsay Hi There
+ __________ 
+< Hi There >
+ ---------- 
+        \   ^__^
+         \  (oo)\_______
+            (__)\       )\/\
+                ||----w |
+                ||     ||
 ~~~
 
 <!--more-->
 
+
+
 ### Useful Options / Examples
+There are many useful options for this intersting cowsay command.
+The complete list is: cowsay  `[-e  eye_string]` `[-f cowfile]` `[-h]` `[-l]` `[-n]` `[-T tongue_string]` `[-W column]` `[-bdgpstwy]`
+
+`-e` allows the user to customize the cow's eyes which are defaulted to "oo" as seen above.
+`-T` allows the user to add a tongue string to the cow image. By default there is no tongue string.
+`-W` allows the user to customize the number of character printed in the message before the message wraps around
+`-[-bdgpstwy]` are all options to define different pre-made cows emotional states. For example, adding the -g flag will result in a greedy cow to print and -s will result in a stoned cow to print.
 
-#### Example command
+#### cowsay [-bdgpstwy] [`user_input`]
+
+~~~ bash
+$ cowsay -s hi there
+ __________ 
+< hi there >
+ ---------- 
+        \   ^__^
+         \  (**)\_______
+            (__)\       )\/\
+             U  ||----w |
+                ||     ||
+~~~
 
 ##### Break it down
+There are several provided modes which change the appearance of the cow depending on its particular emotional/physical state
+The user can set this state using these flags
+####cowsay [-f cowfile] [`user_string`]
```

Comment:
Notice that with `jekyll serve` running (as the assignment describes), the
page will automatically rebuild every time you make an edit. You should be
able to just play around a bit to get the formatting to look pretty.

> On Mar 23 2016, at 5:22 pm, Alex Chojnacki &lt;notifications@github.com&gt;
wrote:  

>

> In [_commands/toys/cowsay.md](https://github.com/c4cs/c4cs.github.io/pull/96
#discussion_r57239470):

>  
>  
>     >  
>     >  ##### Break it down

>     > +There are several provided modes which change the appearance of the
cow depending on its particular emotional/physical state

>     > +The user can set this state using these flags

>     > +####cowsay [-f cowfile] [`user_string`]

>  
>

> Yeah, when I was mentioning that 'newlines can help' I should have been more
specific. I mean that when you get the weird formatting errors like a header
being displayed with the # in the line, just moving it to the next line
usually resolves the rendering issue.

>

> —  
You are receiving this because you are subscribed to this thread.  
Reply to this email directly or [view it on GitHub](https://github.com/c4cs/c4
cs.github.io/pull/96/files/c8fe36e731a610e6cff724ad7fd6a3977c73b117#r57239470)
![](https://github.com/notifications/beacon/AAUt3vLZH6OtQoDQZ8fRsuaZl52VbChkks
5pwa8ZgaJpZM4H01Se.gif)